megamek/src/megamek/server/Server.java

```diff
@@ -887,6 +887,16 @@ private void receivePlayerVersion(Packet packet, int connId) {
             System.out.println("ERROR: Client/Server Version Mismatch -- Client: "+version+" Server: "+MegaMek.VERSION);
             needs = true;
         }
+        else if (clientChecksum == null) {    // print a message indicating client doesn't have jar file
```

Comment:
Yes, I overlooked that before.  You're right, the version and checksum checks should be separate.  Additionally, there should be a new-line check, like when the checksums don't match.